runners/spark/src/main/java/org/apache/beam/runners/spark/io/SourceDStream.java

```diff
@@ -112,6 +128,10 @@ public String name() {
     return "Beam UnboundedSource [" + id() + "]";
   }
 
+  public int getNumPartitions() {
```

Comment:
Notes: "exposing the number of partitions for this SourceDStream so we can set the appropriate partitioner on the read via `mapWithState`..." or something.